external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaTokenSparkConf.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.kafka010
+
+import org.apache.kafka.clients.CommonClientConfigs
+import org.apache.kafka.common.config.{SaslConfigs, SslConfigs}
+import org.apache.kafka.common.security.auth.SecurityProtocol.SASL_SSL
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.Logging
+
+private[spark] case class KafkaTokenClusterConf(
+  identifier: String,
+  bootstrapServers: String,
+  targetServersRegex: String,
+  securityProtocol: String,
+  kerberosServiceName: String,
+  trustStoreLocation: Option[String],
+  trustStorePassword: Option[String],
+  keyStoreLocation: Option[String],
+  keyStorePassword: Option[String],
+  keyPassword: Option[String],
+  tokenMechanism: String) {
+  override def toString: String = s"KafkaTokenClusterConf{" +
```

Comment:
It's a case class, you get `toString` for free.

Reply:
True but then the passwords are then printed as plaintext. Because of that decided to override.